end-to-end-test.sh

```diff
@@ -154,7 +178,11 @@ get() {
 
 sleep 1
 
-get "127.0.0.1:${port}/metrics" | grep -E -v "${skip_re}" > "${tmpdir}/e2e-output.txt"
+if [ ${socket} -ne 0 ]; then
+  curl -s -X GET --unix-socket "${unix_socket}" ./metrics | grep -E -v "${skip_re}" > "${tmpdir}/e2e-output.txt"
+else
+  get "127.0.0.1:${port}/metrics" | grep -E -v "${skip_re}" > "${tmpdir}/e2e-output.txt"
+fi
```

Comment:
to not duplicate grep, we can do:

    (
      if [ ${socket} -ne 0 ]; then
        curl -s -X GET --unix-socket "${unix_socket}" ./metrics
      else
        get "127.0.0.1:${port}/metrics" 
      fi
    ) | grep -E -v "${skip_re}" > "${tmpdir}/e2e-output.txt"